siliconcompiler/schema/schema_cfg.py

```diff
@@ -3506,6 +3484,177 @@ def schema_constraint(cfg):
+            constraints supported.""")
+
+    scparam(cfg, ['server', 'queue'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Queue selection",
+            switch="-server_queue <str>",
+            example= ["cli: -server_queue nightrun",
+                      "api: chip.set('server', 'queue', 'nightrun')"],
+            schelp="""
+            Schedule the job for the specified queue. With slurm, this
+            translates to 'partition'. The queue name must match
+            the name of an existing job schemduler queue.""")
+
+    scparam(cfg, ['server', 'nice'],
+            sctype='int',
+            scope='job',
+            shorthelp="Server: Scheduling priority",
+            switch="-server_nice <int>",
+            example= ["cli: -server_nice 100",
+                    "api: chip.set('server', 'nice', '100')"],
+            schelp="""
+            Specifies the scheduling priority for the job. Negative numbers
+            increases priority while positive numbers decrease priority.
+            For more information, see the job scheduler documentation.""")
+
+    scparam(cfg, ['server', 'timeout'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Timeout value",
+            switch="-server_timeout <str>",
+            example= ["cli: -server_timeout 16:00",
+                    "api: chip.set('server', 'timeout', '16:00')"],
+            schelp="""
+            Specifies the limit on the total job run time.
+            Acceptable slurm time formats include "minutes", "minutes:seconds",
+            "hours:minutes:seconds", "days-hours", "days-hours:minutes",
+            and "days-hours:minutes:seconds".""")
+
+    scparam(cfg, ['server', 'interactive'],
+            sctype='bool',
+            scope='job',
+            shorthelp="Server: Request interactive session",
+            switch="-server_interactive <str>",
+            example= ["cli: -server_interactive true",
+                    "api: chip.set('server', 'interactive', True)"],
+            schelp="""
+            Specifies that job is interactive. In slurm, this would
+            translate to something like '-i --pty bash'.""")
+
+    scparam(cfg, ['server', 'defer'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Start time",
+            switch="-server_defer <str>",
+            example= ["cli: -server_defer 16:00",
+                    "api: chip.set('server', 'defer', '16:00')"],
+            schelp="""
+            Defer initiation of job until the specified time. The parameter
+            is pass through string for remote job scheduler such as slurm.
+            For more information abotut the exact format specification, see
+            the job scheduler documentation. Examples of valid slurm specific
+            values include: now+1hour, 16:00, 010-01-20T12:34:00.""")
+
+    scparam(cfg, ['server', 'option'],
```

Comment:
It feels a bit odd to accept an arbitrary list of options, when we have dedicated schema parameters for some options such as `--constraint`, resource limits, etc.

Reply:
Perhaps, but what is the alternative? 1.) Be completely prescriptive and only allow the options that we specify 2.) Support all the slurm options (which would be too much). https://slurm.schedmd.com/srun.html 3.) support all options for all schedulers (lsf, sge, slurm). 

It may be that --constraint went too far?

Like with the pnr flow ('var') the idea here is to offer 'experts'/'advanced users' pass through their own variables at their own risk, while making the most common standardized (queue, cores, mem)